src/host/utf8handling.c

```diff
@@ -0,0 +1,33 @@
+/**
+ * \file   utf8handking.c
+ * \brief  Handles conversions between Unicode (UTF-8) and system native encoding (wide chars on Windows)
+ * \author Copyright (c) 2017 Jérôme Leclercq and the Premake project
+ */
+
+#include "premake.h"
+#include "stdlib.h"
+
+#ifdef PLATFORM_WINDOWS
+const char* utf8_fromwide(lua_State* L, const wchar_t* wstr)
```

Comment:
I agree that that is not preferable... In general I was just hoping that a malloc for this isn't something we need... mostly because these match functions are used ALOT, and so have quite a bit of performance impact... on our project we're already hitting 2GB of memory use.

if that removes the need for this method entirely, then I'm OK with just calling it in place to be entirely honest, and simply use stack for this...